packages/core/src/repository/data-source/network.data-source.ts

```diff
@@ -0,0 +1,102 @@
+import { lastValueFrom, Observable } from 'rxjs';
+import { DataSource } from './data-source';
+import { Query } from '../query/query';
+import { MethodNotImplementedError, QueryNotSupportedError } from '../errors';
+import { DataSourceMapper } from './data-source-mapper';
+import { ArrayMapper, JsonDeserializerMapper, VoidMapper } from '../mapper/mapper';
+import { Type } from '../../helpers';
+import {
+    DeleteNetworkQuery,
+    GetNetworkQuery,
+    NetworkQuery,
+    PostNetworkQuery,
+    PutNetworkQuery,
+} from '../query/network-query';
+import { ApiRequestService } from './api-request.service';
+import { HttpRequestBuilder } from '../../data';
+
+export class NetworkDataSource implements DataSource<unknown> {
+    constructor(private readonly requestService: ApiRequestService) {}
+
+    public get(query: Query): Promise<unknown> {
+        if (query instanceof GetNetworkQuery) {
+            return lastValueFrom(this.getRequestWithParameters<unknown>(query).get());
```

Comment:
**Nit:** I don't think this is needed, this kind of generic is usually inferred from the context. **You can do the same in all the cases**. I've tried to remove it… at the compiler is OK with it.

I'm not sure 100% why it works though. I think it's because we've `unknown` as the return type.

```suggestion
            return lastValueFrom(this.getRequestWithParameters(query).get());
```